var/spack/repos/builtin/packages/uncrustify/package.py

```diff
@@ -10,22 +10,47 @@ class Uncrustify(Package):
     """Source Code Beautifier for C, C++, C#, ObjectiveC, Java, and others."""
 
     homepage = "http://uncrustify.sourceforge.net/"
-    url      = "http://downloads.sourceforge.net/project/uncrustify/uncrustify/uncrustify-0.61/uncrustify-0.61.tar.gz"
-
-    version('0.67', sha256='54f15c8ebddef120522db21f38fac1dd3b0a285fbf60a8b71f9e333e96cf6ddc')
+    git      = "https://github.com/uncrustify/uncrustify"
+    url      = "https://sourceforge.net/projects/uncrustify/files/uncrustify/uncrustify-0.69/uncrustify-0.69.tar.gz"
+
+    version('master', branch='master')
+    version('0.74', commit='62048b')
+    version('0.73', commit='25b765')
+    version('0.72', commit='1d3d8f')
+    version('0.71', commit='64d82f')
+    version('0.70', commit='51f64d')
+    version('0.69', commit='a7a8fb')
+    version('0.68', commit='86bc34')
+    version('0.67', commit='00321a')
+    version('0.66', commit='80f549')
+    version('0.65', commit='905676')
+    version('0.64', commit='1d7d97')
+    version('0.63', commit='44ce0f')
+    version('0.62', commit='5987f2')
     version('0.61', sha256='1df0e5a2716e256f0a4993db12f23d10195b3030326fdf2e07f8e6421e172df9')
 
     depends_on('cmake', type='build', when='@0.64:')
+    depends_on('automake', type='build', when='@0.63')
+    depends_on('autoconf', type='build', when='@0.63')
 
-    @when('@0.64:')
+    @when('@0.64:,master')
```

Comment:
Is there a reason you added `master` as an upper bound here? I think in our version ordering, `master < develop`, so for example if a `develop` branch was added here, this would not have a defined `install` method. Other than that this looks great (as for your note on supporting different build systems, the way you did it here is generally how it has to be done - we are aiming to make that easier in the future though).

Reply:
No I just misunderstood how the `@when` function/macro works and thought I need to add the version that is not a number explicitly.
So if I understood it correctly now, the correct way would be to put
```python
@when('@0.64,develop')
```
right?
This means if a version is `>=0.64` or any of the special git branches `master, develop, main, head trunk` use this function. Is it correct?

Reply:
Re note on build systems: I see. The only thing the user "looses" in this case is the information how much time has been spent in which stage because all of the stages (re/configure, build, make, install) are combined into one stage install. Of course it is not really important but it seems kind of misleading to me.

Reply:
> This means if a version is >=0.64 or any of the special git-branch-names master, develop, main, head, trunk use this function. Is it correct?

Yes, although for our version comparisons `develop > any-numbered-version`, so `@64:,develop` is the same as `@64:`. Likewise master/main/head/trunk are all considered greater than any numbered version. Did you encounter issues trying to install `@develop` where Spack wasn't using this function?

Reply:
No everything worked. Again misunderstood the versioning scheme. Removed it now, and in my opinion would be ready to merge now.